nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/source/SecretsDirConfigSource.java

```diff
@@ -97,28 +103,32 @@ public String getValue(String property) {
         String result = properties.get(property);
         if (result != null) {
             try {
-                // check the last modified time
+                // check existence (secret mount might have gone away) and the last modified time
                 FileTime ft = storedModifiedTimes.get(property);
-                Path path = Paths.get(secretsDir.toString(), property);
+                Path path = storedPaths.get(property);
```

Comment:
to be thread-safe `get` should be avoided and `computeIfAbseent` should take its place.

Reply:
"Noice". Didn't knew about `computeIfAbsent`.

Reply:
Outdated, as the properties are stored in an inner class now.